packages/valory/skills/agent_performance_summary_abci/models.py

```diff
@@ -0,0 +1,175 @@
+    )
+
+
+@dataclass
+class AgentPerformanceSummary:
+    """
+    Agent performance summary.
+
+    - If the agent has any activity, fields will be filled.
+    - Otherwise, initial state with nulls and empty arrays.
+    """
+
+    timestamp: Optional[int] = None  # UNIX timestamp (in seconds, UTC)
+    metrics: List[AgentPerformanceMetrics] = field(default_factory=list)
+    agent_behavior: Optional[str] = None
+
+
+class AgentPerformanceSummaryParams(BaseParams):
+    """Agent Performance Summary's parameters."""
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        """Initialize the parameters' object."""
+        self.coingecko_olas_in_usd_price_url: str = self._ensure(
+            "coingecko_olas_in_usd_price_url", kwargs, str
+        )
+        self.store_path: Path = self.get_store_path(kwargs)
+        super().__init__(*args, **kwargs)
+
+    def get_store_path(self, kwargs: Dict) -> Path:
+        """Get the path of the store."""
+        path = self._ensure("store_path", kwargs, str)
+        # check if path exists, and we can write to it
+        if (
+            not os.path.isdir(path)
+            or not os.access(path, os.W_OK)
+            or not os.access(path, os.R_OK)
+        ):
+            raise ValueError(
+                f"Policy store path {path!r} is not a directory or is not writable."
+            )
+        return Path(path)
+
+
+class SharedState(BaseSharedState):
+    """Keep the current shared state of the skill."""
+
+    abci_app_cls: Type[AbciApp] = AgentPerformanceSummaryAbciApp
+
+    @property
+    def params(self) -> AgentPerformanceSummaryParams:
+        """Return the params."""
+        return cast(AgentPerformanceSummaryParams, self.context.params)
+
+    @property
+    def synced_timestamp(self) -> int:
+        """Return the synchronized timestamp across the agents."""
+        return int(
+            self.context.state.round_sequence.last_round_transition_timestamp.timestamp()
+        )
+
+    def _read_existing_performance_summary(self) -> AgentPerformanceSummary:
+        """Read the existing agent performance summary from a file."""
+        file_path = self.params.store_path / AGENT_PERFORMANCE_SUMMARY_FILE
+
+        try:
+            with open(file_path, "r") as f:
+                existing_data = AgentPerformanceSummary(**json.load(f))
+            return existing_data
+        except (FileNotFoundError, json.JSONDecodeError) as e:
+            self.context.logger.warning(
+                f"Could not read existing agent performance summary: {e}"
+            )
+            return AgentPerformanceSummary()
+
+    def _overwrite_performance_summary(self, summary: AgentPerformanceSummary) -> None:
+        """Write the agent performance summary to a file."""
+        file_path = self.params.store_path / AGENT_PERFORMANCE_SUMMARY_FILE
+
+        with open(file_path, "w") as f:
+            json.dump(asdict(summary), f, indent=4)
```

Comment:
Why are these protected since they are accessed? Eg here:

https://github.com/valory-xyz/trader/blob/ee7da746c8353d8aca3e27af9f9193acc84afc88/packages/valory/skills/agent_performance_summary_abci/behaviours.py#L265